### PR TITLE
Fix for audio turn processing and add support for missing properties

### PR DESCRIPTION
**Pull request by modnarb**
- Fix for audio turns being corrupted from reco result returning after turn end
- Add fromId support
- Add outputformat support, add to query params
- Protect against OOM when starting turns before the previous one completed

### CODE REVIEW

**Comment by gghizila** [comment]
Can you change the description of this PR?

File: src/common.speech/AgentConfig.ts [hunk 1]
```diff
@@ -28,7 +28,8 @@ export interface IAgentConfig {
     botInfo: {
         commType: string,
         connectionId: string,
-        conversationId: string
+        conversationId: string,
+        fromId: string
```
Comment (gghizila) [inline src/common.speech/AgentConfig.ts]: I am not sure where is this used.

Comment (modnarb) [reply src/common.speech/AgentConfig.ts]: It's a property that can be set by the customer. WebChat requested it as did the user in this post https://github.com/microsoft/cognitive-services-speech-sdk-js/issues/103

**Comment by modnarb** [comment]
> Can you change the description of this PR?

Done

**Comment by modnarb** [comment]
Re-ran build and all tests pass, #20191206.3

**Pull request merged**